src/main/java/jenkins/plugins/jobcacher/ArbitraryFileCache.java

```diff
@@ -454,6 +456,21 @@ public enum CompressionMethod {
                 GzipCompressorInputStream::new,
                 ".tgz")
         ),
+        TARGZ_BEST_SPEED(new TarArbitraryFileCacheStrategy(
```

Comment:
Have you some examples which shows the difference in speed and archive size if using these parameters compared to the default settings?

Reply:
The best comparison I have found is this one:  https://medium.com/linuxstories/comparison-of-gzip-bzip2-and-xz-compression-tools-7348ed910c68
See the "Testing gzip" section.

Reply:
I was thinking if it would make sense to set this for the `TARGZ` directly, as I did not really thought about this when while implementing the cache compression ;) 
Do you know what happens, if you try to uncompress an archive with different settings as the ones used while compressing? If it works fine, I would say we change the settings of `TARGZ`. Or do you think it would be nice to keep both? 

btw: Thanks for your contribution!

Reply:
The gzip library is able to uncompress gz-files written with different compression levels without further input parameters. The (necessary) metadata about the compression level is written to the gz-file.
In the Java library `commons-compress:1.21` (that this plugin use), the metadata about compression level is written by the class `GzipCompressorOutputStream` at lines 100-106.

For text files, such as source code (the `node_modules` directory in web-apps for instance), it makes sense to use the "default" compression level (which is 6), because it is a "sweet spot" between compression speed and size. My recommendation is not to change that.

For directories with lots of mixed file types, both lots of text and binary files, users might have "need for speed", and can use `TARGZ_BEST_SPEED` :)

For directories with lots of binary files, the `TAR` option might be best.

Reply:
Ok nice, then lets go like this. Thanks for clarifying ;)

Reply:
would you mind to add this additional info to the readme? I think it would be helpful for new users to decide with setting they should choose.

Reply:
I've added a new section **Choosing the compression method** to the README file.
I also included descriptions of the existing options, and I had to guess a bit... so please proof read it and correct me if I'm wrong :)